ir/src/lib.rs

```diff
@@ -24,6 +24,7 @@ pub struct AirIR {
     periodic_columns: Vec<Vec<u64>>,
     boundary_constraints: BoundaryConstraints,
     transition_constraints: TransitionConstraints,
+    public_inputs: Vec<(String, usize)>,
```

Comment:
nit: I would move this above the constraints, probably above the `periodic_columns`